### PR TITLE
Feature/timelines filter custom fields 3701

### PR DESCRIPTION
**Pull request by nkenneweg**
Implements [3701](https://www.openproject.org/work_packages/3701)

### CODE REVIEW

**Comment by GarstgerUnhold** [comment]
Stuff's a-failin'. And looks like it can be related to this, so it's worth checking out before merging this.

**Comment by Nilos** [comment]
I can not see the error-output on travis. Any Idea what is failing?

**Comment by apfelfabrik** [comment]
Merging with dev branch, and testing on internal CI.

File: app/views/timelines/_custom_fields.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<%
+available_filter_types = ["bool", "list"]
+
+custom_fields.each do |custom_field|
+  if available_filter_types.include? custom_field.field_format then
+%>
+  <p>
+    <%= label_tag "timeline_options_custom_fields_#{custom_field.id}", custom_field.name %>
+    <% if custom_field.field_format == "bool" then %>
```
Comment (apfelfabrik) [inline app/views/timelines/_custom_fields.html.erb:9]: This field_format case select looks like it might as well be extracted into separate partials.

**Comment by apfelfabrik** [comment]
CI is green, I hope travis is of the same opinion. I found a problem with filters for (none), though, which is documented in the ticket.

**Pull request merged**